### DISCUSSION TITLE
Try-except doesn't work as intended (failing my try on purpes but it doesnt do the except)

### DISCUSSION THREAD

**Discussion by elnatan2604** [seleniumbase/SeleniumBase#2173] [Q&A] [answered]
I'm not sure if this problem is a SB problem but I'll try anyway. I have this code:             try:
                self.send_keys(fn.FieldsNames.event_html, fn.FieldsNames.position_change)
                self.click(f"//*[contains(text(),'{fn.FieldsNames.position_change}')]")
                f.Functions.wait_for_loading(self)
                self.send_keys(fn.FieldsNames.event_reason_html_job_info, fn.FieldsNames.department_change)
                self.click(f"//*[contains(text(),'{fn.FieldsNames.department_change}')]")
                self.clear(fn.FieldsNames.department_html)
                self.send_keys(fn.FieldsNames.department_html, random_department)
                self.sleep(2)
                self.click(f"//*[contains(text(),'{random_department}')]")
                f.Functions.wait_for_loading(self)
                old_department = self.get_text(fn.FieldsNames.old_department_value_html)
                self.click(f"//*[contains(text(),'{fn.FieldsNames.save}')]")
                self.wait_for_element_absent(f"//*[contains(text(),'{fn.FieldsNames.save}')]")
                self.refresh()
            except:
                self.refresh()
           
and I fail on purpes this line self.send_keys(fn.FieldsNames.department_html, random_department) instead of random_department I wrote "fail test" but my code doesnt fail (or raising exception) so it kinda passes above the except and continues to run the code. what could be the problem.

**Comment by mdmintz** [comment, marked as answer]
That looks like a Python question and not a SeleniumBase one. (Also, your code should be formatted as code with proper line indentation so that it makes sense.)

Here's how a try/except block works:

```python
try:
    raise Exception("Fail!")
except Exception:
    print("Caught!")
```

After the exception is caught in the `try` block, the program continues from the `except` block.
